### PR TITLE
Completed task 6

### PR DESCRIPTION
**Pull request by JasonDMLa**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -101,3 +101,41 @@ describe("GET/api/articles", () => {
...
+      .then(({ body: { comments } }) => {
+        expect(comments.length > 0)
+        expect(comments).toHaveLength(11)
+        comments.forEach((comment) => {
+          expect(comment).toHaveProperty("comment_id");         
+          expect(comment).toHaveProperty("votes");
+          expect(comment).toHaveProperty("created_at");
+          expect(comment).toHaveProperty("author");
+          expect(comment).toHaveProperty("body");
+          expect(comment).toHaveProperty("article_id");
```
Comment (nc-christian-dutton) [inline __tests__/app.test.js:119]: You should know what the article_id is

File: __tests__/app.test.js [hunk 2]
```diff
@@ -101,3 +101,41 @@ describe("GET/api/articles", () => {
...
+});
+
+test("404: returns a corresponding message when given a syntax valid article id isnt found", () => {
+  return request(app)
+    .get("/api/articles/500/comments")
+    .expect(404)
+    .then(({ body }) => {
+      expect(body.msg).toBe("id not found");
+    });
+});
```
Comment (nc-christian-dutton) [inline __tests__/app.test.js:141]: One more test you can do - what if an article does exist but has no comments yet? We would want a 200 and an empty array - not an error. You will have to check that the article exists

File: db/controllers/topics.controller.js [hunk 3]
```diff
@@ -39,23 +40,34 @@ exports.getArticleById = (request, response, next) => {
...
+    .then((idCounts) => {
+      retrieveAllArticles(idCounts)
+        .then((articles) => {
+          const countsToMap = {};
+          idCounts.forEach((articleCount) => {
+            countsToMap[articleCount.article_id] = Number(articleCount.total);
+          });
+          articles.forEach((article) => {
+            article.comment_count = countsToMap[article.article_id] || 0;
+          });
```
Comment (nc-christian-dutton) [inline db/controllers/topics.controller.js:53]: You should be getting the comment_count from within the query in the model - have a look at COUNT.

File: db/models/topics.models.js [hunk 4]
```diff
@@ -85,3 +85,31 @@ exports.retrieveAllArticles = () => {
...
+  .then((comments)=>{
+    return comments.rows
+  })
+}
+
+exports.retrieveCommentsById = (article_id) => {
+  let queryString = "SELECT comment_id,votes,created_at,author,body,article_id FROM comments";
+  const queryValues = [];
+  const queryPromises = [];
+  if (article_id) {
```
Comment (nc-christian-dutton) [inline db/models/topics.models.js:100]: Surely there's always going to be an article_id in some form? Do you need this if statement?
